fe/fe-core/src/main/java/org/apache/doris/nereids/properties/ChildOutputPropertyDeriver.java

```diff
@@ -258,8 +258,35 @@ public PhysicalProperties visitPhysicalHashJoin(
 
         // broadcast
         if (rightOutputProperty.getDistributionSpec() instanceof DistributionSpecReplicated) {
-            DistributionSpec parentDistributionSpec = leftOutputProperty.getDistributionSpec();
-            return new PhysicalProperties(parentDistributionSpec);
+            DistributionSpec leftDistributionSpec = leftOutputProperty.getDistributionSpec();
+            if (hashJoin.child(0).getOutputSet().containsAll(hashJoin.getOutputSet())) {
+                return new PhysicalProperties(leftDistributionSpec);
+            } else if (leftDistributionSpec instanceof DistributionSpecHash
+                    && !(hashJoin.isMarkJoin() && hashJoin.getHashJoinConjuncts().isEmpty())
+                    && !hashJoin.getHashConjunctsExprIds().first.isEmpty()
+                    && !hashJoin.getHashConjunctsExprIds().second.isEmpty()
+                    && hashJoin.getHashConjunctsExprIds().first.size()
+                        == hashJoin.getHashConjunctsExprIds().second.size()
+                    && hashJoin.getHashConjunctsExprIds().first.containsAll(
+                            ((DistributionSpecHash) leftDistributionSpec).getOrderedShuffledColumns())) {
+                List<ExprId> leftIds = hashJoin.getHashConjunctsExprIds().first;
+                List<ExprId> rightIds = hashJoin.getHashConjunctsExprIds().second;
+                List<ExprId> usedRightExprIds = Lists.newArrayList();
+                for (ExprId exprId : ((DistributionSpecHash) leftDistributionSpec).getOrderedShuffledColumns()) {
```

Comment:
conjuncts expr id may in `EquivalenceSet`

Reply:
done